transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -108,7 +108,8 @@ public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgum
         throw new IllegalArgumentException(
           "Field " + config.field + " is not present in the input schema");
       } else {
-        if (!inputSchemaField.getSchema().getType().equals(Schema.Type.STRING)) {
+        if (!inputSchemaField.getSchema().getType().equals(Schema.Type.STRING) &&
```

Comment:
lets move the schema validation logic into a package private method and add a unit test. `validateSchema(Schema inputSchema)`